.github/workflows/web-vercel-deploy.yml

```
@@ -0,0 +1,54 @@
+name: Web Vercel deploy
+on:
+  workflow_call:
+    inputs:
+      environment:
+        description: 'The environment to deploy to'
+        default: 'Test'
+        required: false
+        type: string
+      vercel_org_id:
+        description: 'The vercel organisation ID'
+        required: true
+        type: string
+      vercel_project_id:
+        description: 'The ID of the vercel project'
+        required: true
+        type: string
+    secrets:
+      npm_pat:
+        description: The PAT for NPM registry access
+        required: true
+      vercel_auth_token:
+        description: The vercel auth token for vercel access
+        required: true
+
+jobs:
+  deploy_vercel:
+    environment: ${{ inputs.environment }}
+    runs-on: ubuntu-latest
+    name: 'Preview deploy'
+    env:
+      ENVIRONMENT: ${{ inputs.environment }}
+    strategy:
+      matrix:
+        node-version: ['18.x']
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Setup node
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{ matrix.node-version }}
+          registry-url: https://npm.pkg.github.com
+          cache: yarn
+      - name: Install dependencies
+        run: yarn install --production=false --frozen-lockfile
+        env:
+          NODE_AUTH_TOKEN: ${{secrets.NPM_PAT}}
```

Can we do `run` last? It makes it easier to read 
```suggestion
        env:
          NODE_AUTH_TOKEN: ${{secrets.NPM_PAT}}
        run: yarn install --production=false --frozen-lockfile
```